// terra-i18n/lib/intlLoaders.js
'use strict';

var loadArIntl = function loadArIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/ar.js');
  }, 'ar-intl-local');
};

var loadEnIntl = function loadEnIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/en.js');
  }, 'en-intl-local');
};

var loadEnGBIntl = function loadEnGBIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/en-GB.js');
  }, 'en-GB-intl-local');
};

var loadEnUSIntl = function loadEnUSIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/en-US.js');
  }, 'en-US-intl-local');
};

var loadDeIntl = function loadDeIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/de.js');
  }, 'de-intl-local');
};

var loadPtIntl = function loadPtIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/pt.js');
  }, 'pt-intl-local');
};

var loadPtBRIntl = function loadPtBRIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/pt-BR.js');
  }, 'pt-BR-intl-local');
};

var loadFrIntl = function loadFrIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/fr.js');
  }, 'fr-intl-local');
};

var loadFrFRIntl = function loadFrFRIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/fr-FR.js');
  }, 'fr-FR-intl-local');
};

var loadEsIntl = function loadEsIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/es.js');
  }, 'es-intl-local');
};

var loadEsUSIntl = function loadEsUSIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/es-US.js');
  }, 'es-US-intl-local');
};

var loadEsESIntl = function loadEsESIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/es-ES.js');
  }, 'es-ES-intl-local');
};

var loadFiFIIntl = function loadFiFIIntl() {
  return require.ensure([], function (require) {
    return require('intl/locale-data/jsonp/fi-FI.js');
  }, 'fi-FI-intl-local');
};

var intlLoaders = {
  ar: loadArIntl,
  en: loadEnIntl,
  'en-GB': loadEnGBIntl,
  'en-US': loadEnUSIntl,
  de: loadDeIntl,
  pt: loadPtIntl,
  'pt-BR': loadPtBRIntl,
  fr: loadFrIntl,
  'fr-FR': loadFrFRIntl,
  es: loadEsIntl,
  'es-US': loadEsUSIntl,
  'es-ES': loadEsESIntl,
  'fi-FI': loadFiFIIntl
};

module.exports = intlLoaders;
